internal/helm/chart/dependency_manager_test.go

```diff
@@ -496,54 +772,90 @@ func TestDependencyManager_addRemoteDependency(t *testing.T) {
 func TestDependencyManager_resolveRepository(t *testing.T) {
 	tests := []struct {
 		name                       string
-		repositories               map[string]*repository.ChartRepository
-		getChartRepositoryCallback GetChartRepositoryCallback
+		downloaders                map[string]CleanableDownloader
+		getChartDownloaderCallback GetChartDownloaderCallback
 		url                        string
-		want                       *repository.ChartRepository
-		wantRepositories           map[string]*repository.ChartRepository
+		want                       CleanableDownloader
+		wantDownloaders            map[string]CleanableDownloader
 		wantErr                    string
 	}{
 		{
-			name: "resolves from repositories index",
+			name: "resolves from downloaders index",
 			url:  "https://example.com",
-			repositories: map[string]*repository.ChartRepository{
-				"https://example.com/": {URL: "https://example.com"},
+			downloaders: map[string]CleanableDownloader{
+				"https://example.com/": &repository.ChartRepository{URL: "https://example.com"},
 			},
 			want: &repository.ChartRepository{URL: "https://example.com"},
 		},
 		{
 			name: "resolves from callback",
 			url:  "https://example.com",
-			getChartRepositoryCallback: func(url string) (*repository.ChartRepository, error) {
+			getChartDownloaderCallback: func(_ string) (CleanableDownloader, error) {
 				return &repository.ChartRepository{URL: "https://example.com"}, nil
 			},
 			want: &repository.ChartRepository{URL: "https://example.com"},
-			wantRepositories: map[string]*repository.ChartRepository{
-				"https://example.com/": {URL: "https://example.com"},
+			wantDownloaders: map[string]CleanableDownloader{
+				"https://example.com/": &repository.ChartRepository{URL: "https://example.com"},
 			},
 		},
 		{
 			name: "error from callback",
 			url:  "https://example.com",
-			getChartRepositoryCallback: func(url string) (*repository.ChartRepository, error) {
+			getChartDownloaderCallback: func(_ string) (CleanableDownloader, error) {
 				return nil, errors.New("a very unique error")
 			},
-			wantErr:          "a very unique error",
-			wantRepositories: map[string]*repository.ChartRepository{},
+			wantErr:         "a very unique error",
+			wantDownloaders: map[string]CleanableDownloader{},
 		},
 		{
 			name:    "error on not found",
 			url:     "https://example.com",
 			wantErr: "no chart repository for URL",
 		},
+		{
+			name: "resolves from oci repository",
+			url:  "oci://example.com",
+			downloaders: map[string]CleanableDownloader{
+				"oci://example.com/": &repository.OCIChartRepository{URL: url.URL{
+					Scheme: "oci",
+					Host:   "example.com",
+				},
+				},
```

Comment:
Some formatting inconsistencies in how these brackets are closed here and below.